graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/RemoteReindexingMigrationAdapterOS2.java

```diff
@@ -117,6 +119,8 @@ public class RemoteReindexingMigrationAdapterOS2 implements RemoteReindexingMigr
     private final DatanodeRestApiProxy datanodeRestApiProxy;
     private final NotificationService notificationService;
 
+    private final DatanodeMigrationLockService migrationLockService;
```

Comment:
As we now know that this service contains locks for one GL node only... can't it contain locks for one migration only?
I am thinking about not injecting it as a singleton object, but creating it from scratch as an object belonging to each migration.

What it could change : this class would not have to know the locks and pass it between different methods.
It could just ask ITS OWN service to "release all locks aquired during this run".

What do you think?

Reply:
The problem is that locking needs to happen from both sides - from the migration and from the rotation & retention tasks. No migration of an index set can start if one of the maintenance tasks operates on that index set. So it has to be global in one graylog node from the service perspective and global over the whole cluster through the mongodb distributed approach. It's not enough to manage locks in the migration itself. The maintenance tasks would then need to ask migrations if there is any running and holds any locks. Or do I understand the suggestion wrong?

Reply:
Yeah, you are right, I have just focused on releasing aspect, forgetting about the other aspects...